src/Analysis.cpp

```diff
@@ -360,9 +361,17 @@ void Analysis::EmitCollapsedTemplates()
 void Analysis::EmitCollapsedTemplateOpt()
 {
     std::unordered_map<std::string, InstantiateEntry> collapsed;
+    std::unordered_map<DetailIndex, std::string> collapsedNameCache;
+    
     for (const auto& fn : functions)
     {
-        auto &stats = collapsed[collapseName(llvm::demangle(GetBuildName(fn.first.first)))];
```

Comment:
I guess this was showing up in profiles?

When writing this, it did cross my mind that I was doing potentially redundant demangling.  I didn't cache the demangling at the time though, because I was worried about the memory consumption.  If that looks fine on Unity, then it looks fine to me.

Reply:
It was showing up a bit indeed (not much). I've checked that the produced results are the same in all tests & in the full unity build analysis, so Should Be Fine :)